File: api/web/models/chat_message.py
"""APIs for ChatMessage"""
from typing import Dict, Optional, Any, List
from modules.types import PMObject


class ChatMessageInstance(PMObject):
    id: int
    created_at: str
    uuid: str

    role: str
    name: Optional[str] = None
    content: Optional[str] = None
    tool_calls: Optional[List[Dict[str, Any]]] = None
    function_call: Optional[Dict[str, Any]] = None

    token_count: Optional[int] = None

    chat_message_metadata: Optional[Dict[str, Any]] = None

    session_uuid: str


class ChatLogViewInstance(PMObject):
    project_uuid: str
    chat_model_uuid: str
    chat_model_name: str

    chat_model_version_uuid: str
    chat_model_version: int

    session_uuid: str
    assistant_message_uuid: str

    created_at: str

    user_input: str
    assistant_output: Optional[str] = None
    tool_calls: Optional[List[Dict[str, Any]]] = None
    function_call: Optional[Dict[str, Any]] = None

    latency: Optional[float] = None
    cost: Optional[float] = None
    prompt_tokens: Optional[int] = None
    completion_tokens: Optional[int] = None
    total_tokens: Optional[int] = None

    run_from_deployment: bool


class ChatLogsCountInstance(PMObject):
    project_uuid: str
    count: int

File: src/types/ChatMessage.ts
// Interfaces for ChatMessage retrieval processes

/**
 * Interface for request to read ChatModelVersion's ChatMessageSessions.
 */
export interface ReadSessionChatMessagesRequest {
  chat_session_uuid: string;
  page: number;
  rows_per_page: number;
}

/**
 * Interface for request to read Project's ChatMessageSessions.
 */
export interface ReadProjectChatMessagesRequest {
  project_uuid: string;
  page: number;
  rows_per_page: number;
}

/**
 * Interface for request to read Project's ChatMessages count.
 */
export interface ReadProjectChatMessagesCountRequest {
  project_uuid: string;
}

/**
 * Interface for response of ChatMessages count.
 */
export interface ReadProjectChatMessagesCountResponse {
  project_uuid: string;
  count: number;
}

/**
 * General interface for ChatMessage.
 */
export interface ChatMessage {
  id?: number;
  created_at: string;
  uuid?: string;
  role: string;
  name?: string;
  content: string;
  tool_calls?: Record<string, any>;
  tokens_count?: number;
  latency?: number;
  cost?: number;
  chat_log_metadata?: Record<string, any>;
  session_uuid?: string;
}
